src/ClientWidgetApi.ts

```diff
@@ -477,21 +477,31 @@ export class ClientWidgetApi extends EventEmitter {
             });
         }
 
-        const isState = request.data.state_key !== null && request.data.state_key !== undefined;
-        let sendEventPromise: Promise<ISendEventDetails>;
-        if (isState) {
-            if (!this.canSendStateEvent(request.data.type, request.data.state_key!)) {
+        let sendEventPromise: Promise<ISendEventDetails|ISendFutureDetails>;
+        if (request.data.state_key != null) {
```

Comment:
This is intentional, because it checks for `null` and `undefined` at the same time.  If that's not obvious enough, I can change it to be explicit with `... !== null && ... !== undefined`.